umake/frameworks/ide.py

```diff
@@ -473,3 +474,115 @@ def post_install(self):
+        preg = re.compile(".*/images_www/v6/download/.*")
+        for line in url_version_str.split("\n"):
+            if preg.match(line):
+                line = line.replace("var PAGE_ARTIFACTS_LOCATION = \"/images" + \
+                                    "_www/v6/download/", "").replace("/\";", "")
+                return line.strip()
+        #Fallback
+        logger.error("Could not determine latest version, using fallback...")
+        return "8.0.2"
+      
+    def download_provider_page(self):
+        """Get things prepared for the download."""
+        logger.info("Fetching latest version...")
+        self.version = self.get_latest_version()
+        logger.info("Netbeans {}".format(self.version))     
+        #FIXME Replace urlopen with something from DownloadCenter.
+        logger.info("Fetching download url...")
+        url_file = urlopen("https://netbeans.org/images_www/v6/download/" \
+                           "{}/js/files.js".format(self.version))
+        #Extract the file from the response                 
+        url_file_str = url_file.read().decode('utf-8')
+        url_string = self.search_url_string(url_file_str, self.version, self.flavour)
+        string_array = url_string.split(", ")
+        url_suffix = string_array[0]
+        md5 = string_array[2]
+        self.complete_name = "netbeans{}".format(self.flavour)
+
+        download_url = "{}{}{}{}".format(self.BASE_URL, self.version, "/final/", url_suffix)
+        
+        @MainLoop.in_mainloop_thread
+        def done(download_result):
+            #FIXME Why is it trying to download the md5 instead of just checking it?
+            """
+            ERROR: 9c7a9c16644d3508068ad47d4f467c4f couldn't finish download: 
+            Invalid URL '9c7a9c16644d3508068ad47d4f467c4f': No schema supplied. Perhaps you meant http://9c7a9c16644d3508068ad47d4f467c4f?
+            """
+            self.download_requests.append(DownloadItem(download_url, Checksum(ChecksumType.md5, md5)))
+            self.start_download_and_install()
+
+        DownloadCenter(urls=[DownloadItem(md5, None)], on_done=done, download=False)
+    
+    @property
+    def is_installed(self):
+        # check path and requirements
+        if not super().is_installed:
+            return False
+        if not isfile(join(self.install_path, self.executable)):
+            logger.debug("{} binary isn't installed".format(self.name))
+            return False
+        return True
+        
+    def post_install(self):
+        """Create the Netbeans launcher"""
+        icon_path = self.install_path + 'nb/netbeans'
+        exec_path = self.install_path + 'bin/netbeans'
+        comment = _("Netbeans IDE " + self.version)
+        categories = "Development;IDE;"
+        create_launcher(self.desktop_filename,
+                        get_application_desktop_file(name=_("Netbeans IDE " \
+                                                     + self.version),
```

Comment:
No version in the name itself please. If the framework can later autoupdate, it's something we don't want to support (especially if it's upstream's autoupdating tool). The promise we give is to always download latest stable version.